src/main/java/Katheryne.java

```diff
@@ -0,0 +1,91 @@
+import java.util.*;
+
+public class Katheryne {
+    public static void main(String[] args) throws KatheryneExceptions {
+        ArrayList<Task> lst = new ArrayList<Task>();
+        System.out.println("Ad astra abyssosque! I am Katheryne, the receptionist here at the Adventurers' Guild.");
+        System.out.println("How may I assist?");
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            try {
+                String userInput = in.nextLine();
+                String[] keywordInput = userInput.split(" ", 2);
+                if (userInput.equalsIgnoreCase("BYE")) {
+                    System.out.println("Ad Astra Abyssoque, Traveller!");
+                    break;
+                } else if (keywordInput[0].equalsIgnoreCase("done") || keywordInput[0].equalsIgnoreCase("complete")) {
+                    if (keywordInput.length == 1) {
+                        throw new KatheryneExceptions("Traveller, please specify by index which task you completed.");
+                    }
+                    int i = Integer.parseInt(keywordInput[1]);
+                    if (i <= lst.size() && i > 0) {
+                        lst.get(i - 1).markAsDone();
+                        System.out.println(lst.get(i - 1));
+                    } else {
+                        throw new KatheryneExceptions("That's not a valid index...");
+                    }
+                } else if (keywordInput[0].equalsIgnoreCase("delete")) {
+                    if (keywordInput.length == 1) {
+                        throw new KatheryneExceptions("Traveller, please specify by index what I should delete.");
+                    }
+                    int i = Integer.parseInt(keywordInput[1]);
+                    if (i <= lst.size() && i > 0) {
+                        System.out.println("Okay, I'll delete the following item:");
+                        System.out.println(lst.remove(i - 1));
+                    } else {
+                        throw new KatheryneExceptions("That's not a valid index...");
+                    }
+                } else if (userInput.equalsIgnoreCase("list")) {
+                    System.out.println("Here's the list I've stored for you:");
+                    for(int i = 1; i <= lst.size(); i++) {
+                        System.out.println(i + ") " + lst.get(i - 1));
+                    }
+                } else if (keywordInput[0].equalsIgnoreCase("deadline")) {
+                    if (keywordInput.length == 1) {
+                        throw new KatheryneExceptions("A deadline needs a description and a /by time to complete it");
+                    }
+                    String[] parsedDeadline = keywordInput[1].split(" /by ");
+                    if (parsedDeadline.length == 2) {
+                        lst.add(new Deadline(parsedDeadline[0], parsedDeadline[1]));
+                        System.out.println("'" + parsedDeadline[0] + "' added to your list, do by " + parsedDeadline[1]);
```

Comment:
This line seems to be more than 120 characters and violates the style guide. Perhaps try to break up the text into two lines?